pkg/ddevapp/ddevapp.go

```diff
@@ -1594,31 +1594,54 @@ func FindNotOmittedImages(app *DdevApp) []string {
 	return images
 }
 
-// FindMaxTimeout looks through all services and returns the max timeout found
-// Defaults to 120
-func (app *DdevApp) FindMaxTimeout() int {
-	const defaultContainerTimeout = 120
-	maxTimeout := defaultContainerTimeout
+// GetMaxContainerWaitTime looks through all services and returns the max time we expect
+// to wait for all containers to become `healthy`. Mostly this is healtcheck.start_period.
+// Defaults to DefaultContainerTimeout (usually 120 unless overridden)
+func (app *DdevApp) GetMaxContainerWaitTime() int {
+	defaultContainerTimeout, _ := strconv.Atoi(app.DefaultContainerTimeout)
+	maxWaitTime := defaultContainerTimeout
+
 	if app.ComposeYaml == nil {
 		return defaultContainerTimeout
 	}
 	if y, ok := app.ComposeYaml["services"]; ok {
 		for _, v := range y.(map[string]interface{}) {
 			if i, ok := v.(map[string]interface{})["healthcheck"]; ok {
-				if timeout, ok := i.(map[string]interface{})["timeout"]; ok {
-					duration, err := time.ParseDuration(timeout.(string))
+				if startPeriod, ok := i.(map[string]interface{})["start_period"]; ok {
+					duration, err := time.ParseDuration(startPeriod.(string))
 					if err != nil {
 						continue
 					}
 					t := int(duration.Seconds())
-					if t > maxTimeout {
-						maxTimeout = t
+					if t > maxWaitTime {
+						maxWaitTime = t
+					}
+				} else { /* In this case we didn't have a specified start_period, so guess at one */
+					interval := 1
+					retries := 3
+					// Use a generic default for start_period/maxWaitTime based on
+					// `retries` (3) * default `interval` (5)
```

Comment:
I believe it's `start_interval`, not `interval`, and let's add a reference to where it comes from.

```suggestion
					// https://docs.docker.com/reference/dockerfile/#healthcheck
					// `retries` (3) * default `start_interval` (5)
```